src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -33,56 +30,80 @@ public void setValid(boolean valid) {
         this.valid = valid;
     }
 
-    public List<String> getMessages() {
-        return messages;
+    public List<String> getErrorMessages() {
+        return errorMessages;
     }
 
-    public void setMessages(List<String> messages) {
-        this.messages = messages;
-    }
-
-    public void addMessage(String message) {
-        if (messages == null) {
-            messages = new ArrayList<String>();
+    public void addErrorMessage(String message) {
+        if (errorMessages == null) {
+            errorMessages = new ArrayList<String>();
         }
-        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(message)) {
-            messages.add(message);
+        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(errorMessages, message)) {
+            errorMessages.add(message);
         }
     }
     
-    private boolean isNotDuplicateMessage(String message) {
-        for (String currentMessage : getMessages()) {
+    private boolean isNotDuplicateMessage(List<String> messages, String message) {
+        for (String currentMessage : messages) {
             if (StringUtils.equalsIgnoreCase(currentMessage, message)) {
                 return false;
             }
         }
         return true;
     }
 
-    public void addMessages(List<String> messagesToAdd) {
-        if (CollectionUtils.isNotEmpty(messagesToAdd)) {
-            for (String messageToAdd : messagesToAdd) {
-                addMessage(messageToAdd);
+    public void addErrorMessages(List<String> errorMessagesToAdd) {
+        if (CollectionUtils.isNotEmpty(errorMessagesToAdd)) {
+            for (String messageToAdd : errorMessagesToAdd) {
+                addErrorMessage(messageToAdd);
             }
         }
     }
 
     public String getErrorMessagesAsOneFormattedString() {
+        return buildOneFormattedString(errorMessages);
+    }
+    
+    private String buildOneFormattedString(List<String> messageList) {
         StringBuffer result = new StringBuffer();
-
-        if (messages != null && !messages.isEmpty()) {
-            for (String message : messages) {
+        if (messageList != null && !messageList.isEmpty()) {
+            for (String message : messageList) {
                 result.append(message);
                 result.append(KFSConstants.NEWLINE);
             }
         }
-        
         return result.toString();
-    }   
+    }
+    
+    public List<String> getDetailMessages() {
+        return detailMessages;
+    }
     
+    public void addDetailMessage(String message) {
+        if (detailMessages == null) {
+            detailMessages = new ArrayList<String>();
+        }
```

Comment:
Is this null check necessary?

Reply:
it followed suit for how the error messages are handled.  I don't think it should happen, but I would hate for the job to fail due to a preventable NPE.